openmetrics.go

```diff
@@ -0,0 +1,209 @@
+package openmetrics
+
+import (
+	"sort"
+	"sync"
+)
+
+// ContentType is the official content type of an openmetrics document.
+const ContentType = "application/openmetrics-text; version=1.0.0; charset=utf-8"
+
+// MetricPoint is a point in a Metric.
+type MetricPoint struct {
+	Suffix   MetricSuffix
+	Value    float64
+	Label    Label
+	Exemplar *Exemplar
+}
+
+// MetricType defines the type of a Metric.
+type MetricType uint8
+
+func (t MetricType) String() string {
+	switch t {
+	case GaugeType:
+		return "gauge"
+	case CounterType:
+		return "counter"
+	case StateSetType:
+		return "stateset"
+	case InfoType:
+		return "info"
+	case HistogramType:
+		return "histogram"
+	case _GaugeHistogramType:
+		return "gaugehistogram"
+	case _SummaryType:
+		return "summary"
+	default:
+		return "unknown"
+	}
+}
+
+const (
+	// UnknownType must use unknown MetricPoint values.
+	UnknownType MetricType = iota
+	// GaugeType must use gauge MetricPoint values.
+	GaugeType
+	// Counter must use counter MetricPoint values.
+	CounterType
+	// StateSetType set must use stateset MetricPoint values.
+	StateSetType
+	// InfoType must use info MetricPoint values.
+	InfoType
+	// HistogramType must use histogram MetricPoint values.
+	HistogramType
+	// GaugeHistogramType must use gaugehistogram value MetricPoint values.
+	_GaugeHistogramType
+	// Summary quantiles must use summary value MetricPoint values.
+	_SummaryType
```

Comment:
Why these are prefixed with `_` (Kinda private? Shouldn't they be exposed anyway?)

Reply:
no, because they don't really exist in this lib. Summary should not be used at all and GaugeHistogram has no meaning for Go